src/main/java/org/spongepowered/mod/registry/SpongeGameRegistry.java

```diff
@@ -1406,20 +1414,69 @@ private void setTextFactory() {
     }
 
     private void setSelectors() {
-        /*
-         * try { SelectorTypes.class.getDeclaredField("ALL_PLAYERS").set(null,
-         * new SpongeSelectorType("a"));
-         * SelectorTypes.class.getDeclaredField("ALL_ENTITIES").set(null, new
-         * SpongeSelectorType("e"));
-         * SelectorTypes.class.getDeclaredField("NEAREST_PLAYER").set(null, new
-         * SpongeSelectorType("p"));
-         * SelectorTypes.class.getDeclaredField("RANDOM_PLAYER").set(null, new
-         * SpongeSelectorType("r")); } catch (Exception e) {
-         * e.printStackTrace(); } RegistryHelper.setFactory(SelectorTypes.class,
-         * new SpongeSelectorTypeFactory());
-         * RegistryHelper.setFactory(Selectors.class, new
-         * SpongeSelectorFactory());
-         */
+        RegistryHelper.mapFields(SelectorType.class, this.selectorMappings);
+        SpongeSelectorFactory factory = new SpongeSelectorFactory();
+        try {
+            // POSITION
+            ArgumentType<Integer> x = factory.createArgumentType("x", Integer.class);
+            ArgumentType<Integer> y = factory.createArgumentType("y", Integer.class);
+            ArgumentType<Integer> z = factory.createArgumentType("z", Integer.class);
+            ArgumentHolder.Vector3<Vector3i, Integer> position = new SpongeArgumentHolder.SpongeVector3<Vector3i, Integer>(x, y, z, Vector3i.class);
+            ArgumentTypes.class.getDeclaredField("POSITION").set(null, position);
```

Comment:
It may be neater if these are put in a field mapping, `Map<String, ArgumentHolder>` and then use `RegistryHelper.mapFields` at the end.